RFS/src/main/java/org/opensearch/migrations/metadata/GlobalMetadataCreatorResults.java

```diff
@@ -1,14 +1,22 @@
 package org.opensearch.migrations.metadata;
 
 import java.util.List;
+import java.util.stream.Stream;
 
 import lombok.Builder;
 import lombok.Data;
 
 @Builder
 @Data
 public class GlobalMetadataCreatorResults {
-    private List<String> legacyTemplates;
-    private List<String> indexTemplates;
-    private List<String> componentTemplates;
+    private List<CreationResult> legacyTemplates;
+    private List<CreationResult> indexTemplates;
+    private List<CreationResult> componentTemplates;
+
+    public long fatalIssueCount() {
+        return Stream.of(getLegacyTemplates(), getIndexTemplates(), getComponentTemplates())
+            .flatMap(List::stream)
+            .filter(r -> !r.wasFatal())
```

Comment:
IIRC, filter includes elements that are true, so this will yield the nonFatalIssueCount

Reply:
Nice catch, added a test case too.